### PR TITLE
SDQL to QueryRepo

### PR DESCRIPTION
**Pull request by astrimaitis**
You can use Test-harness to test the Questionnaire responses.

Running Test-Harness:

Insight Platform Simulator --> Create Campaign
Core --> Opt In To Campaign
Insight Platform Simulator --> Upload Questionnaire
Insight Platform Simulator --> Post Query --> Query 2 (q5 asks for questionnaire, check i5 in response for response) --> Approve Query

### CODE REVIEW

File: packages/persistence/src/volatile/FakeDBVolatileStorage.ts [hunk 1]
```diff
@@ -11,7 +11,17 @@ import {
   VolatileStorageKey,
   VolatileStorageMetadata,
 } from "@snickerdoodlelabs/objects";
-import { indexedDB as fakeIndexedDB } from "fake-indexeddb";
+import { indexedDB as fakeIndexedDB, IDBCursor,
```
Comment (SnickerChar) [inline packages/persistence/src/volatile/FakeDBVolatileStorage.ts:14]: Seems like a lot of these imports are not used

File: packages/query-parser/src/interfaces/objects/AST_QuestionnaireQuery.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    static fromSchema(
+      name: SDQL_Name,
+      schema: ISDQLQueryClause,
+    ): AST_QuestionnaireQuery {
+      return new AST_QuestionnaireQuery(
+        name,
+        ESDQLQueryReturn.Object,
+        web2QueryTypes[10],
```
Comment (SnickerChar) [inline packages/query-parser/src/interfaces/objects/AST_QuestionnaireQuery.ts:41]: Where does this 10 come from? Is there an enum we can use instead to reference this?

File: packages/query-parser/src/interfaces/objects/SDQLQueryWrapper.ts [hunk 3]
```diff
@@ -194,6 +209,10 @@ export class SDQLQueryWrapper {
     return this.internalObj.insights;
   }
 
+  public getQuestionSchema(): ISDQLQuestionBlock[] {
```
Comment (SnickerChar) [inline packages/query-parser/src/interfaces/objects/SDQLQueryWrapper.ts:212]: What part of SDQL has questions?

File: packages/query-parser/src/sampleData/questionnaireQuery.data.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  insights: {
+    i1: {
+      name: "Questionnaire",
+      target: "$q1",
+      returns: "'qualified'",
+    },
+  },
+  compensations: {
+    parameters: {
```
Comment (SnickerChar) [inline packages/query-parser/src/sampleData/questionnaireQuery.data.ts:29]: Should clean up the compensations block for this query

File: packages/test-harness/src/mocks/InsightPlatformSimulator.ts [hunk 5]
```diff
@@ -438,6 +506,35 @@ export class InsightPlatformSimulator {
       });
   }
 
+  public uploadQuestionnaire(): ResultAsync<
+    IpfsCID, Error
+  > {
+    const questionnaireData = JSON.stringify(
```
Comment (SnickerChar) [inline packages/test-harness/src/mocks/InsightPlatformSimulator.ts:512]: Can we import this from the sample data directly instead of copying it?

**Review by SnickerChar — COMMENTED** [review]
Lots of cleanup to do. Run the diff yourself, there's a lot of files changed that are not for formating purposes, things like unused imports.

**Review by SnickerChar — DISMISSED** [review]
LGTM!

**Review by orhunerd — APPROVED** [review]

**Review by OzanEm — APPROVED** [review]

**Pull request merged**